### PR TITLE
Added xkcd Style for Markers (plot only)

### LINKED ISSUE #2625: Markers in XKCD style
Currently, XKCD style is not applied on markers in a plots (e.g. created with plot(x,y,'o') or scatter(X,Y,marker='s')). It would be nice if circles, squares, crosses etc. looks like they are hand drawn.

### PR DESCRIPTION
**Pull request by TarasKuzyo**
## PR Summary
Partially fixes #2625 (Markers in XKCD style ).

I added missing `path.sketch` parameters when drawing markers in `Line2D`
with the following changes from xkcd defaults: `(scale, length/2, 2*randomness)`.
It allows better sketch visibility on small-sized objects.

The functionality was requested for `plt.plot` only but I expected to add it for `plt.scatter` as well. 
Unlike `plt.plot` makers in `plt.scatter` are drawn with `Collection` object (yet in both cases `renderer.draw_markers` is called).  AFAU `PathEffectRenderer` is required to display xkcd-style drawings properly but `self._paths` (responsible for enabling path effects)  is explicitly set to `None` in`Collection`'s [constructor](https://github.com/matplotlib/matplotlib/blob/e09dece2278b6a9aa34d731ca1567072dea4bbb2/lib/matplotlib/collections.py#L167). When I tried commenting that line out, I got the following error message:

>   File "/home/taras/Project/Activity/matplotlib/lib/matplotlib/patheffects.py", line 206, in draw_path
>     Stroke.draw_path(self, renderer, gc, tpath, affine, rgbFace)
>   File "/home/taras/Project/Activity/matplotlib/lib/matplotlib/patheffects.py", line 195, in draw_path
>     renderer.draw_path(gc0, tpath, trans, rgbFace)
>   File "/home/taras/Project/Activity/matplotlib/lib/matplotlib/backends/backend_agg.py", line 161, in draw_path
>     self._renderer.draw_path(gc, path, transform, rgbFace)
> SystemError: new style getargs format but argument is not a tuple

It has something to do with incorrect argument passed to some internal function, 
yet I am not sure how to proceed with it further.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jklymak** [comment]
You're going to have to change the xkcd test image.  See `lib/matplotlib/tests/test_path.py`

**Comment by TarasKuzyo** [comment]
Ok, thank you. Should I include a marker in the test case as well or should I rather write a separate test for a plot with markers?

**Comment by jklymak** [comment]
I assumed the test case had a marker or the test shouldn't have failed?  Certainly if there was no marker in the old test, I'd go ahead and add one since you'll have to change the image anyway.  Thanks!

**Comment by jklymak** [comment]
OK< so the change you made caused those differences?  They don't seem like a big deal to me, so I think its fine to put the changed image in.  Up to you if you think adding more to that test makes sense or not.  Does the xkcd example look OK after this?  It'd be nice to see a before/after to know exactly what you've done here...

**Comment by TarasKuzyo** [comment]
Yes, still not sure where those changes come from. I will add one more test case then.
Attaching a sample plot with markers
marker-xkcd

**Comment by jklymak** [comment]
:+1:, but what did it look like on master?

**Comment by TarasKuzyo** [comment]
You can see it attached in the issue [#2625](https://github.com/matplotlib/matplotlib/issues/2625)

**Comment by jklymak** [comment]
> The circles should have a slightly deformed but smooth outline  

I suspect that if one of the offset circles was made a different radius that would do the trick...  Not sure how easy that is to impliment...

File: lib/matplotlib/tests/test_path.py [hunk 1]
```diff
@@ -122,6 +122,22 @@ def test_xkcd():
         ax.plot(x, y)
 
 
+@image_comparison(baseline_images=['xkcd_marker'], remove_text=True)
```
Comment (dstansby) [inline lib/matplotlib/tests/test_path.py]: Could you add `extensions=['png']` to this bit, and only generate the `.png` images? It saves us cluttering up the repository with svg and pdf images.

Comment (TarasKuzyo) [reply lib/matplotlib/tests/test_path.py]: @dstansby, added it

**Comment by anntzer** [comment]
I think(?) the trick for "better" circles is likely to play more with the sketch params, e.g. use something based on the markersize, or at least something even smaller than /2 (I agree with @timhoffm they don't look very good).  Perhaps try some values and see if you can come up with something, although it's probably not worth overthinking it either.

**Comment by TarasKuzyo** [comment]
The problem with the circle shape arises because of the path wiggle, so it does not close properly. 
Basically, start and end path nodes do not match. I am not sure if it is possible to come up with easy fix without breaking anything else.

**Comment by NelleV** [comment]
@TarasKuzyo If I am not mistaken, the changes @dstansby asked you to make means you can delete the pdf and svg file.

It also means that whoever merges the pull request will have to use the squash and merge option to actually get rid of the binaries.

**Comment by tacaswell** [comment]
Re-milestoned to 3.1 as part of the 3.0 release triage.

**Comment by TarasKuzyo** [comment]
@NelleV, thanks! I think I've done it in the last commit. Yet, I am not sure regarding next steps.

**Comment by NelleV** [comment]
@TarasKuzyo We wait until CI has finished, and until @dstansby re-reviews this PR.

**Comment by NelleV** [comment]
The pdf and svg files are still here on teh other hand.

**Comment by TarasKuzyo** [comment]
@NelleV, do you mean I had to delete pdfs and svgs for both xkcd tests?

**Comment by NelleV** [comment]
I believe that the current test only rely on pngs, so yes, I'd remove the pdfs and svgs as they are not relevant anymore.

**Comment by NelleV** [comment]
Great! Thanks @TarasKuzyo !

**Pull request merged**